sunkit_instruments/goes_xrs/goes_chianti_tem.py

```diff
@@ -0,0 +1,212 @@
+
+__all__ = ["goes_calculate_temperature_em", "_goes_chianti_temp_em", "_manage_goesr_detectors"]
+
+
+def goes_calculate_temperature_em(goes_ts, abundance="coronal", remove_scaling=False):
+    """
+    This calculates the temperature and emission measure from the GOES/XRS flux ratios.
+
+    Parameters
+    ----------
+    goes_ts : `~sunpy.timeseries.XRSTimeSeries`
+        The GOES XRS timeseries containing the data of both the xrsa and xrsb channels (in units of W/m**2).
+    abundance: `~str`, {`coronal` | `photospheric`}
+        Default 'coronal'. Which abundances to use for the calculation.
+    remove_scaling: `Boolean`
+        Checks whether to remove the SWPC scaling factors. Default False for "true" fluxes from the netcdf files. 
+        This is only an issue and will need to be used for the older FITS files for 8-15 XRS.
+
+    Returns
+    -------
+    `~sunpy.timeseries.GenericTimeSeries`
+        Conatins the temperature and emission measure calculated from the input `goes_ts` TimeSeries.
+
+    Example
+    -------
+    >>> goes_ts = ts.TimeSeries("sci_xrsf-l2-flx1s_g16_d20170910_v2-1-0.nc").truncate("2017-09-10 12:00", "2017-09-10 20:00")
+    >>> output = goes_calculate_temperature_em(goes_ts)
+
+    """
+    if not isinstance(goes_ts, ts.XRSTimeSeries):
+        raise TypeError(f"goes_ts must be a XRSTimeSeries object, not {type(goes_ts)}")
+
+    if goes_ts.observatory is None:
+        raise ValueError(f"The GOES satellite number is not found in the goes_ts")
+    else:        
+        satellite_number = int(goes_ts.observatory.split("-")[-1])
+        if (satellite_number <1) or (satellite_number>17):
+            raise ValueError(f"The GOES satellite number has to be between 1 and 17, {satellite_number} is given")
+
+    # check if GOES-R and whether the primary detector values are given
+    if (satellite_number>=16):  
+        if "primary_detector_a" in goes_ts.columns:
+            output = _manage_goesr_detectors(goes_ts, satellite_number, abundance=abundance)
+        else:
+            warn_user("No information about primary/secondary detectors in XRSTimeSeries, assuming primary for all")
+            output = _goes_chianti_temp_em(goes_ts, satellite_number, abundance=abundance)
+
+    # check if the older files are passed
+    elif ("Origin" in goes_ts.meta.metas[0]) and (goes_ts.meta.metas[0]["Origin"] == "SDAC/GSFC"):
+        output = _goes_chianti_temp_em(goes_ts, satellite_number, abundance=abundance, remove_scaling=True)
+
+    else: 
+        output = _goes_chianti_temp_em(goes_ts, satellite_number, abundance=abundance)
+    
+    return output
+
+
+@manager.require('goes_chianti_response_table',
+                ['https://hesperia.gsfc.nasa.gov/ssw/gen/idl/synoptic/goes/goes_chianti_response_latest.fits'],
+                 '4ca9730fb039e8a04407ae0aa4d5e3e2566b93dfe549157aa7c8fc3aa1e3e04d')
+
+def _goes_chianti_temp_em(goes_ts, satellite_number, secondary=0, abundance="coronal", remove_scaling=False):
```

Comment:
```suggestion
def _chianti_temp_emiss(goes_ts, satellite_number, secondary=0, abundance="coronal", remove_scaling=False):
```
I don't think we need the word goes in the function call.